src/board_builder/test/accuracy/accuracy_test.py

```diff
@@ -0,0 +1,224 @@
+            diff = np.array(arr1) - np.array(arr2)
+            rms_error += np.mean(diff ** 2)
+            total_points += 1
+
+        rms_error = np.sqrt(rms_error / total_points)
+        return rms_error
+
+    def run_accuracy_tester(self):
+        module = importlib.import_module(f"src.board_builder.test.accuracy.board_definitions.{self._parameters.SCENE_NAME}")
+        board_definition = getattr(module, "BOARD_DEFINITION")
+
+        ### REFERENCE DATA ###
+        for pose in self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE:
+            self.board_builder.pose_solver.set_intrinsic_parameters(pose.target_id, self._parameters.DETECTOR_INTRINSICS)
+
+        self.board_builder.pose_solver.set_detector_poses(self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE)
+
+        ### GENERATE SNAPSHOTS OF THE SCENE ###
+        snapshots = generate_virtual_snapshots(board_definition, self._parameters.NUMBER_OF_SNAPSHOTS)
+        average_difference = self._calculate_generate_snapshots_inaccuracy(board_definition, snapshots)
+        print(f"Generated snapshots inaccuracy (average Frobenius norm difference): {average_difference}")
+
+        ### COLLECTION DATA ###
+        two_dimension_collection_data = []
+        predicted_corners = []
+
+        for snapshot in snapshots:
+            collection_data = generate_data(snapshot, self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE, remove_markers_out_of_frame=True)
+            two_dimension_collection_data.append(collection_data)
+            noisy_collection_data = self._add_noise_to_corners(collection_data)
+            collection_corners = self.board_builder.collect_data(noisy_collection_data)
+            predicted_corners.append(collection_corners)
+        #graph_renderer(snapshots, two_dimension_collection_data, self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE)
+
+        ### BUILD BOARD ###
+        predicted_board = self.board_builder.build_board(repeatability_testing=False)
+
+        # Center the board definition around the reference marker
+        board_definition_np = {k: np.array(v) for k, v in board_definition.items()}
+        marker_0_corners = board_definition_np[0]
+        center_marker_0 = np.mean(marker_0_corners, axis=0)
+        simulated_board_definition = {str(k): v - center_marker_0 for k, v in board_definition_np.items()}
+
+        simulated_markers = [
+            Marker(marker_id=k, points=v.tolist())
+            for k, v in simulated_board_definition.items()
+        ]
+        simulated_board = TargetBoard(target_id='simulated_board', markers=simulated_markers)
+
+        # RMS
+        # TODO: FIX RMS ERROR
+        #  If we compare predicted board vs simulated board, the y and z value error is lower than 0.5, but the x value
+        #  seems to be consistently off by around 6 mm, which is where the error is mainly coming from. I can't find
+        #  why that is right now.
```

Comment:
Is this still the case? If so, let's step through the code in a debugger together on Monday and try to isolate the source.

Reply:
Yes, The predicted target pose seems to have an offset (around 6 mm) in the x direction of the world reference system. I tried to investigate this issue and narrowed it down a bit. This might be a problem in either the projection (input data), pose solver (algorithm), or detector intrinsics (calibration)